### ISSUE TITLE
Bug when using optimizer and WarmupLR togather

### ISSUE THREAD

**Issue by wjw136** [deepspeedai/DeepSpeed#7303]
During I use deepspeed, I set the learning rate of AdamW optimizer at 1e-5, but this setting will be overrided by the default **max\_lrs** values (0.001) from the scheduler implementation.
It will very influence the training process.

The following is the location of WarmupLR scheduler from the deepspeed source code:
https://github.com/deepspeedai/DeepSpeed/blob/0e3209a16ba595f7a9f5249dffdeb3cf9e420ce4/deepspeed/runtime/lr_schedules.py#L666

The following is my config file:

```
{
    "bf16": { # bf16牺牲精度换取范围，一定不会溢出。
      "enabled": true,
    },
    "zero_optimization": {
        "stage": 1,
        "allgather_partitions": true,
        "allgather_bucket_size": 2e8,
        "overlap_comm": true,
        "reduce_scatter": true,
        "reduce_bucket_size": 2e8,
    },
    "optimizer": {
        "type": "AdamW",
        "params": {
            "lr": 5e-7,
            "betas": [
                0.8,
                0.999
            ],
            "eps": 1e-8,
            "weight_decay": 3e-7
        }
    },
    "scheduler": {
        "type": "WarmupLR",
        "params": {
            "warmup_num_steps": 1
        }
    },
    "gradient_accumulation_steps": 16,
    "train_micro_batch_size_per_gpu": 1,
    "gradient_clipping": 1.0,

    # 日志相关
    "comms_logger": {
        "enabled": false           # 禁用 DeepSpeed 的通信日志
    },
    "environment": {
        "NCCL_DEBUG": "WARN"       # 覆盖 NCCL 日志级别
    },

    # 必须模型先拆出来
    # "pipeline": {
    #     "pipeline_parallel_degree": 2, 
    #     "activation_checkpointing": true 
    # }

    # 激活检查点
    "activation_checkpointing": {
        "partition_activations": true, 
        "contiguous_memory_optimization": true, 
        "cpu_checkpointing": true, 
        "number_checkpoints": 100, 
        "synchronize_checkpoint_boundary": true, 
        "profile": false 
    },
    "logging": {
        "rank_zero_only": true,
        "level": "INFO"
    },
    "steps_per_print": 1
    "ddp_timeout": 36000,
}
```
**Although I set the learning rate to 5e-7, but the actual learning rate is 1e-3.**
I think this is a crucial bug for normal usage.

**Comment by Flink-ddd** [comment]
Hi@wjw136
Thank you for ask this question, I can take a look at this issue. I'll be working on a fix.

**Comment by tjruwase** [comment]
Completed by #7360
